From b945d57a5acd12bda320a63eb9e45bbb7586cdde Mon Sep 17 00:00:00 2001
From: Aaron Smith <aaron@soccergeek.net>
Date: Mon, 16 Dec 2024 11:48:33 -0800
Subject: [PATCH] Fix "invalid conversion from 'int' to 'CURLoption'" error

Use cast to 'Curloption' to fix compiler error regarding invalid conversion from 'int' to 'CURLoption'.

Upstream: https://github.com/jpbarrette/curlpp/pull/178
Signed-off-by: Thomas Bonnefille <thomas.bonnefille@bootlin.com>
---
 include/curlpp/Options.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/curlpp/Options.hpp b/include/curlpp/Options.hpp
index 40b64ed..292eaa7 100644
--- a/include/curlpp/Options.hpp
+++ b/include/curlpp/Options.hpp
@@ -308,7 +308,7 @@ namespace options
 	typedef curlpp::OptionTrait<long, CURLOPT_LOW_SPEED_LIMIT> LowSpeedLimit;
 	typedef curlpp::OptionTrait<long, CURLOPT_LOW_SPEED_TIME> LowSpeedTime;
 	typedef curlpp::OptionTrait<long, CURLOPT_MAXCONNECTS> MaxConnects;
-	typedef curlpp::OptionTrait<curl_closepolicy, CURLOPT_CLOSEPOLICY> ClosePolicy;
+	typedef curlpp::OptionTrait<curl_closepolicy, (CURLoption)CURLOPT_CLOSEPOLICY> ClosePolicy;
 	typedef curlpp::OptionTrait<bool, CURLOPT_FRESH_CONNECT> FreshConnect;
 	typedef curlpp::OptionTrait<bool, CURLOPT_FORBID_REUSE> ForbidReuse;
 	typedef curlpp::OptionTrait<long, CURLOPT_CONNECTTIMEOUT> ConnectTimeout;

